### PR TITLE
65 save grocery cart items in user cart during checkout, JWT fixes, styling fixes

### PR DESCRIPTION
**Pull request by melleg**
Users can now save their cart items in the checkout screen to their actual own account, and will redirect to /login if they are not authenticated. JWTs are properly encoded and decoded. Carousel items now expand to screen width in mobile view.

### CODE REVIEW

File: backend/src/main/java/dev/itvitae/grocerystore/security/SecurityConfig.java [hunk 1]
```diff
@@ -23,55 +23,58 @@
 @EnableWebSecurity
 public class SecurityConfig {
 
-    private final MyUserDetailsService userDetailsService;
-    private final JWTAuthenticationFilter authenticationFilter;
+  private final MyUserDetailsService userDetailsService;
+  private final JWTAuthenticationFilter authenticationFilter;
 
-    private static final String[] SECURED_URLS = {};
+  private static final String[] ADMIM_URLS = {};
```
Comment (bartspreij) [inline backend/src/main/java/dev/itvitae/grocerystore/security/SecurityConfig.java:29]: ADMIN

File: backend/src/main/java/dev/itvitae/grocerystore/security/SecurityConfig.java [hunk 2]
```diff
@@ -23,55 +23,58 @@
...
-            AuthenticationConfiguration authenticationConfiguration) throws Exception {
-        return authenticationConfiguration.getAuthenticationManager();
-    }
+  @Bean
+  public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+    return http.csrf(csrf -> csrf.disable())
+        .authorizeHttpRequests(
+            authz ->
+                authz
+                    .requestMatchers(ADMIM_URLS)
```
Comment (bartspreij) [inline backend/src/main/java/dev/itvitae/grocerystore/security/SecurityConfig.java:48]: ADMEME

**Review by bartspreij — APPROVED** [review]
Nice

**Comment by melleg** [comment]
Have to load Buy Again from User

**Pull request merged**